sdk/go/common/workspace/project_test.go

```diff
@@ -62,6 +64,86 @@ func TestProjectValidation(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestProjectValidationFailsForIncorrectDefaultValueType(t *testing.T) {
+	t.Parallel()
+	project := Project{Name: "test", Runtime: NewProjectRuntimeInfo("dotnet", nil)}
+	invalidConfig := make(map[string]ProjectConfigType)
+	invalidConfig["instanceSize"] = ProjectConfigType{
+		Type:    "integer",
+		Items:   nil,
+		Default: "hello",
+	}
+
+	project.Config = invalidConfig
+	err := project.Validate()
+	assert.Contains(t,
+		err.Error(),
+		"The default value specified for configuration key 'instanceSize' is not of the expected type 'integer'")
+
+	invalidValues := make([]interface{}, 0)
```

Comment:
```suggestion
	invalidValues := [...]interface{} { "hello"} 
```
Inline array initialization is a little nicer than make/append.